File: WorldEditCUI/src/main/scala/jk_5/worldeditcui/network/packet/PacketSelection.java
package jk_5.worldeditcui.network.packet;

import jk_5.worldeditcui.render.*;
import jk_5.worldeditcui.render.region.*;

import scala.*;

/**
 * No description given
 *
 * @author jk-5
 */
public class PacketSelection extends Packet {

    @Override
    protected void setupArguments() {
        this.setNumberOfArguments(1, 1);
    }

    @Override
    public void process() throws PacketException {
        if(this.getStringArgument(0).equals("cuboid")){
            WERenderer$.MODULE$.selection_$eq(Some.<Region>apply(new CuboidRegion()));
        }else if (this.getStringArgument(0).equals("polygon2d")){
            WERenderer$.MODULE$.selection_$eq(Some.<Region>apply(new PolygonRegion()));
        }else if (this.getStringArgument(0).equals("ellipsoid")){
            WERenderer$.MODULE$.selection_$eq(Some.<Region>apply(new EllipsoidRegion()));
        }else if (this.getStringArgument(0).equals("cylinder")){
            WERenderer$.MODULE$.selection_$eq(Some.<Region>apply(new CylinderRegion()));
        }else{
            throw new PacketException("Invalid selection type. Must be cuboid|polygon2d|ellipsoid|cylinder");
        }
    }
}

File: WorldEditCUI/src/main/scala/jk_5/worldeditcui/network/packet/PacketSelection.scala
package jk_5.worldeditcui.network.packet

import jk_5.worldeditcui.render.WERenderer
import jk_5.worldeditcui.render.region._
import scala.Some

/**
  * No description given
  *
  * @author jk-5
  */
class PacketSelection extends Packet(1, 1) {
  override def process() = this.getString(0) match {
    case "cuboid" => WERenderer.selection = Some(new CuboidRegion)
    case "polygon2d" => WERenderer.selection = Some(new PolygonRegion)
    case "ellipsoid" => WERenderer.selection = Some(new EllipsoidRegion)
    case "cylinder" => WERenderer.selection = Some(new CylinderRegion)
    case _ => throw new PacketException("Invalid selection type. Must be cuboid|polygon2d|ellipsoid|cylinder")
  }
}
